src/state/post-movie.ts

```diff
@@ -0,0 +1,38 @@
+import { AnyAction } from 'redux';
+import { ThunkAction, ThunkDispatch } from 'redux-thunk';
+import RootState from '../types/root-state';
+import MovieService from '../services/movies-service';
+import {
+  handleErrorAction,
+  handleLoadingAction,
+  handleMoviesAction,
+  handleStopLoadingAction,
+} from './action-creators';
+import { FormInput } from '../types/form-input';
+
+const postMovie =
+  (
+    data: FormInput,
+    onModalClose: () => void,
+    handleCongratModalOpen: () => void,
+    filter: string,
+    sort: string,
+    offset: number
+  ): ThunkAction<void, RootState, unknown, AnyAction> =>
+  (dispatch: ThunkDispatch<RootState, unknown, AnyAction>) => {
+    dispatch(handleLoadingAction());
+    MovieService.createMovie(data)
+      .then(() => {
+        onModalClose();
```

Comment:
1. Close/open should be in the component.
2. loading should be managed in handleMoviesAction.
3. filter, sort, and offset should be taken from the store.
4. What about handling errors of getMovies actions?